docarray/array/array_stacked.py

```diff
@@ -91,6 +91,20 @@ def _from_columns(
         da_stacked._docs = docs
         return da_stacked
 
+    def to(self, device: str):
+        """Move the data to the given device
+
+        :param device: the device to move the data to
+        """
+        for field in self._columns.keys():
+            col = self._columns[field]
+            if isinstance(col, TorchTensor):
+                self._columns[field] = col.get_comp_backend().to_device(col, device)
+            elif isinstance(col, NdArray):
```

Comment:
are thes isntance checks needed or can we do one check for instance of `AbstractTensor`? the would be cleaner and ensure that we don't forget to update once we add support for more frameworks.

Reply:
This is a simpler example of this issue
```python
from typing import overload, Callable, Union

@overload
def meow(a: int) -> Callable[[int], int]:
    ...

@overload
def meow(a: str) -> Callable[[str], str]:
    ...

def meow(a):
    if isinstance(a, int):
        return lambda x: 1
    if isinstance(a, str):
        return lambda x: 'str'

# OK
a: str = 's'
meow(a)(a)

# Not ok though it should be
s: Union[int, str] = 's'
meow(s)(s)
```
```python
woof.py:23: error: Argument 1 has incompatible type "Union[int, str]"; expected "int"  [arg-type]
woof.py:23: error: Argument 1 has incompatible type "Union[int, str]"; expected "str"  [arg-type]
```

Reply:
Hrmm I see why it doesnt make this connection though. We cannot be guaranteed that `meow` is a pure function. `meow` could change the type of `a` in its implementation and thus assuming `s` is always string in `meow(s)(s)` is invalid.

Reply:
In the same way, for
```python
self._columns[field] = col.get_comp_backend().to_device(col, device)
```

We cannot be guaranteed that `col.get_comp_backend()` wont change the type of `col`?

Reply:
can we open an issue on mypy and linked it in the code ?